docroot/modules/custom/mass_alerts/src/Controller/ApiController.php

```diff
@@ -0,0 +1,237 @@
+    }
+
+    $build = [
+      '#theme' => 'mass_alerts_sitewide',
+      '#emergencyAlerts' => $results['emergencyAlerts'],
+      '#cache' => [
+        'max-age' => 60
+      ],
+    ];
+
+    $output = $this->renderer->renderRoot($build);
+    $response = new Response();
+    $response->setContent($output);
+    return $response;
+  }
+
+  /**
+   * Returns the specific page alerts rendered.
+   */
+  public function handlePageRequest($nid, Request $request) {
+
+    $results = [
+      'headerAlerts' => [],
+    ];
+
+    $nodeStorage = $this->entityTypeManager()->getStorage('node');
+
+    $currentPage = $nodeStorage->load($nid);
+
+    if ($currentPage) {
+      $organizations = $currentPage->get('field_organizations')->getValue();
+      $org_ids = [];
+
+      foreach ($organizations as $org) {
+        $org_ids[] = $org['target_id'];
+      }
+
+      $query = $nodeStorage->getQuery();
+      $query->condition('type', 'alert');
+      $query->condition('status', 1);
+
+      $orContition = $query->orConditionGroup();
+      $orContition->condition('field_target_page.target_id', $nid);
+      $orContition->condition('field_target_organization.target_id', $org_ids, 'IN');
+
+      $query->condition($orContition);
+
+      $nids = $query->execute();
+
+      $nodes = $nodeStorage->loadMultiple(array_values($nids));
+
+      $alerts = [];
+
+      foreach ($nodes as $node) {
+
+        $changed_date = $node->getChangedTime();
+        $id = $node->uuid() . '__' . $changed_date;
+
+        $items = $node->get('field_alert')->referencedEntities();
+        $item = reset($items);
+
+        $link_type = $item->get('field_emergency_alert_link_type')->getString();
+
+        if ($link_type == '1') {
+          $uri = $item->get('field_emergency_alert_link')->getString();
+          if ($uri) {
+            $url = Url::fromUri($uri)->toString();
+          }
+        }
+        else {
+          $url = $node->toUrl()->toString();
+          $url .= '#' . $item->id();
+        }
+
+        $prefix = NULL;
+        $severity = $node->get('field_alert_severity')->getString();
+
+        if ($severity == 'informational_notice') {
+          $prefix = 'Notice';
+        }
+
+        $alert = [
+          'id' => $id,
+          'text' => $item->get('field_emergency_alert_message')->getString(),
+          'href' => ($link_type == '2') ? '' : $url,
+          'info' => '',
+        ];
+
+        if ($alert) {
+          $alert['prefix'] = $prefix;
+        }
+
+        $alerts[$changed_date] = $alert;
+      }
+
+      krsort($alerts);
+      $results['headerAlerts'] = array_values($alerts);
+    }
+
+    $build = [
+      '#theme' => 'mass_alerts_page',
+      '#headerAlerts' => $results['headerAlerts'],
+      '#cache' => [
+        'max-age' => Cache::PERMANENT,
+        'tags' => [
+          'node:' . $nid
+        ]
+      ],
+    ];
+
+    $output = $this->renderer->renderRoot($build);
+    $response = new Response();
```

Comment:
This needs to be \Drupal\Core\Cache\CacheableResponse for both types of alerts.

Reply:
Actually, I think you do call renderRoot() and set the result to value of #markup.